code/game/objects/items/devices/scanners.dm

```diff
@@ -90,168 +99,178 @@ REAGENT SCANNER
 	melt_temperature = MELTPOINT_PLASTIC
 	origin_tech = "magnets=1;biotech=1"
 	var/last_reading = null
-	var/mode = 1;
+	var/verbosity = 1
 
 /obj/item/device/healthanalyzer/attack(mob/living/M as mob, mob/living/user as mob)
 	if(!user.hallucinating())
-		last_reading = healthanalyze(M, user, mode)
+		last_reading = healthanalyze(M, user, verbosity)
 	else
-		if( (M.stat == DEAD || (M.status_flags & FAKEDEATH)) )
-			user.show_message("<span class='game say'><span class='name'>[src]</span> beeps, \"It's dead, Jim.\"</span>",MESSAGE_HEAR,"<span class='notice'>The [name] glows black.</span>")
+		if((M.stat == DEAD || (M.status_flags & FAKEDEATH)))
+			user.show_message("<span class='game say'><b>\The [src] beeps</b>, \"It's dead, Jim.\"</span>", MESSAGE_HEAR ,"<span class='notice'>\The [src] glows black.</span>")
 		else
-			user << "<span class='notice'>[src] glows [pick("red","green","blue","pink")]! You wonder what would that mean.</span>"
+			user << "<span class='notice'>\The [src] glows [pick("red", "green", "blue", "pink")]! You wonder what that would mean.</span>"
 	src.add_fingerprint(user)
 
 /obj/item/device/healthanalyzer/attack_self(mob/living/user as mob)
 	. = ..()
 	if(.)
 		return
+	if(!user.dexterity_check())
+		user << "<span class='warning'>You don't have the dexterity to do this!</span>"
+		return
 	if(last_reading)
-		user.show_message("<span class='bnotice'Accessing Prior Scan Result</span>")
-		user.show_message(last_reading)
+		user << "<span class='bnotice'>Accessing Prior Scan Result</span>"
+		user << last_reading
 
-proc/healthanalyze(mob/living/M as mob, mob/living/user as mob, var/mode = 0, var/skip_checks=0, var/silent=0)
+//Note : Used directly by other objects. Could benefit of OOP, maybe ?
+proc/healthanalyze(mob/living/M as mob, mob/living/user as mob, var/verbosity = 0, var/skip_checks = 0, var/silent = 0)
 	var/message = ""
 	if(!skip_checks)
-		//writepanic("[__FILE__].[__LINE__] \\/proc/healthanalyze() called tick#: [world.time]")
-		if (( (M_CLUMSY in user.mutations) || user.getBrainLoss() >= 60) && prob(50))
-			user << text("<span class='warning'>You try to analyze the floor's vitals!</span>")
-			for(var/mob/O in viewers(M, null))
-				O.show_message(text("<span class='warning'>[user] has analyzed the floor's vitals!</span>"), 1)
-			user.show_message(text("<span class='notice'>Analyzing Results for The floor:\n\t Overall Status: Healthy</span>"), 1)
-			user.show_message(text("<span class='notice'>\t Damage Specifics: [0]-[0]-[0]-[0]</span>"), 1)
-			user.show_message("<span class='notice'>Key: Suffocation/Toxin/Burns/Brute</span>", 1)
-			user.show_message("<span class='notice'>Body Temperature: ???</span>", 1)
+		if(!user.dexterity_check())

```

Comment:
No, don't do this. This proc can be used in many ways this check should not be here.